music21/midi/realtime.py

```diff
@@ -104,29 +106,37 @@ def __init__(self, streamIn, **keywords):
         self.streamIn = streamIn
 
     def play(self, busyFunction=None, busyArgs=None,
-             endFunction=None, endArgs=None, busyWaitMilliseconds=50):
+             endFunction=None, endArgs=None, busyWaitMilliseconds=50, playForMilliseconds=float("inf"), blocked=True):
         '''
         busyFunction is a function that is called with busyArgs when the music is busy every
         busyWaitMilliseconds.
 
         endFunction is a function that is called with endArgs when the music finishes playing.
+
+        playForMilliseconds is the amount of time in milliseconds after which the playback will automatically stopped
+
+        If blocked is False, the method will finish before ending the stream, allowing you to completely control whether to stop it. Ignore every other arguments
         '''
         streamStringIOFile = self.getStringOrBytesIOFile()
-        self.playStringIOFile(streamStringIOFile, busyFunction, busyArgs,
-                              endFunction, endArgs, busyWaitMilliseconds)
+        self.playStringIOFile(streamStringIOFile, busyFunction, busyArgs, playForMilliseconds,
```

Comment:
playForMilliseconds will need to go at the end, in case someone was calling this function by position elements -- better to put at end, and make blocked and playForMilliseconds keyword-only.